### ISSUE TITLE
2.2.20790: AddUnits#buildUnitOwnerMap:46 - IllegalStateException

### ISSUE THREAD

**Issue by tripleabuilderbot** [triplea-game/triplea#7370]
## User Description
end of british move
save game - 100 % reproduceable on load

## Map
world_war_ii_v3

## Log Message
java.lang.IllegalStateException: Duplicate key b1a7076f-ea79-4dcb-b986-b27cf9dda20c (attempted merging values Italians and Italians)

## TripleA Version
2.2.20790

## Java Version
11.0.6

## Operating System
Mac OS X

## Memory
Heap utilization statistics [MB]
Used Memory: 311
Free memory: 1736
Total memory: 2048
Max memory: 2048

## Stack Trace
```
java.lang.IllegalStateException: java.lang.IllegalStateException: Duplicate key b1a7076f-ea79-4dcb-b986-b27cf9dda20c (attempted merging values Italians and Italians)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at games.strategy.triplea.odds.calculator.ConcurrentBattleCalculator.calculate(ConcurrentBattleCalculator.java:234)
	at games.strategy.triplea.ai.pro.util.ProOddsCalculator.callBattleCalc(ProOddsCalculator.java:224)
	at games.strategy.triplea.ai.pro.util.ProOddsCalculator.callBattleCalc(ProOddsCalculator.java:200)
	at games.strategy.triplea.ai.pro.util.ProOddsCalculator.calculateBattleResults(ProOddsCalculator.java:133)
	at games.strategy.triplea.ai.pro.util.ProOddsCalculator.calculateBattleResults(ProOddsCalculator.java:116)
	at games.strategy.triplea.ai.pro.ProNonCombatMoveAi.moveUnitsToBestTerritories(ProNonCombatMoveAi.java:1961)
	at games.strategy.triplea.ai.pro.ProNonCombatMoveAi.doNonCombatMove(ProNonCombatMoveAi.java:152)
	at games.strategy.triplea.ai.pro.ProNonCombatMoveAi.simulateNonCombatMove(ProNonCombatMoveAi.java:69)
	at games.strategy.triplea.ai.pro.AbstractProAi.purchase(AbstractProAi.java:227)
	at games.strategy.triplea.ai.AbstractAi.start(AbstractAi.java:507)
	at games.strategy.engine.framework.ServerGame.waitForPlayerToFinishStep(ServerGame.java:537)
	at games.strategy.engine.framework.ServerGame.runStep(ServerGame.java:407)
	at games.strategy.engine.framework.ServerGame.startGame(ServerGame.java:297)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.launchInternal(LocalLauncher.java:86)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.lambda$launch$0(LocalLauncher.java:60)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Duplicate key b1a7076f-ea79-4dcb-b986-b27cf9dda20c (attempted merging values Italians and Italians)
	at java.base/java.util.stream.Collectors.duplicateKeyException(Collectors.java:133)
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:180)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at games.strategy.engine.data.changefactory.AddUnits.buildUnitOwnerMap(AddUnits.java:46)
	at games.strategy.engine.data.changefactory.AddUnits.<init>(AddUnits.java:32)
	at games.strategy.engine.data.changefactory.ChangeFactory.addUnits(ChangeFactory.java:81)
	at games.strategy.triplea.odds.calculator.BattleCalculator.calculate(BattleCalculator.java:75)
	at games.strategy.triplea.odds.calculator.ConcurrentBattleCalculator.lambda$calculate$4(ConcurrentBattleCalculator.java:222)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)

```

**Comment by tvleavitt** [comment]
Is this a regression? I thought this was fixed.

Thomas Leavitt

**Comment by maphoenix** [comment]
Here is the game
Load, end british turn and it will crash in Italian turn
[broken.tsvg.zip](https://github.com/triplea-game/triplea/files/5078478/broken.tsvg.zip)

**Comment by trevan** [comment]
I can duplicate it.  It looks like it is related with duplicating transport units somehow.

**Comment by trevan** [comment]
The stack trace is different from the one I fixed earlier.  This one has `at games.strategy.triplea.ai.pro.ProNonCombatMoveAi.doNonCombatMove(ProNonCombatMoveAi.java:152)`.  The other one has `at games.strategy.triplea.ai.pro.ProCombatMoveAi.doCombatMove(ProCombatMoveAi.java:90)`.  Looking at the issues I marked as duplicate of the earlier one, most appear to have the `doNonCombatMove` so I should have realized it earlier that there was another bug.
